sdk/core/azure-core/src/datetime.cpp

```diff
@@ -209,7 +209,8 @@ std::string DateTime::ToString(DateFormat format) const
           leftover);
 
       outCursor += 19;
-      if (fracSec != 0)
+      if ((fracSec != 0 && fractionFormat != TimeFractionFormat::Truncate)
+          || fractionFormat == TimeFractionFormat::AllDigits)
```

Comment:
We don't want to write a `.0` if `fracSec` is 0 but `fractionFormat == TimeFractionFormat::AllDigits`.

This condition seems to have changed that behavior now. Can you test out what happens in that case?

Reply:
I added cases for that. AllDigits is how the caller ask for respecting all the decimal part, even if it is 0, so the string returned is .`0000000Z` in that case

Reply:
That is an odd behavior. You are introducing precision that may not exist in the original date time string.

Reply:
Don't see how.
We are just showing the zeros from the right which cpprest considered useless and removed it.
Now someone is asking to show them.
We don't add the values by some calculation, we only show what's there